src/Domain/CoordinateMaps/Interval.hpp

```diff
@@ -1,135 +1,130 @@
-  void pup(PUP::er& p) noexcept;
-
-  bool is_identity() const noexcept { return is_identity_; }
-
- private:
-  friend bool operator==(const Interval& lhs, const Interval& rhs) noexcept;
-
-  double A_{std::numeric_limits<double>::signaling_NaN()};
-  double B_{std::numeric_limits<double>::signaling_NaN()};
-  double a_{std::numeric_limits<double>::signaling_NaN()};
-  double b_{std::numeric_limits<double>::signaling_NaN()};
-  Distribution distribution_{Distribution::Linear};
-  double singularity_pos_{std::numeric_limits<double>::signaling_NaN()};
-  bool is_identity_{false};
-};
-
-inline bool operator!=(const CoordinateMaps::Interval& lhs,
-                       const CoordinateMaps::Interval& rhs) noexcept {
-  return not(lhs == rhs);
-}
-
-}  // namespace domain::CoordinateMaps
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// Defines the class Interval.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <optional>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/CoordinateMaps/Distribution.hpp"
+#include "Utilities/TypeTraits/RemoveReferenceWrapper.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace domain::CoordinateMaps {
+
+/*!
+ * \ingroup CoordinateMapsGroup
+ * \brief Maps \f$\xi\f$ in the 1D interval \f$[A, B]\f$ to \f$x\f$ in the
+ * interval \f$[a, b]\f$ according to a `domain::CoordinateMaps::Distribution`.
+ *
+ * \details The mapping takes a `domain::CoordinateMaps::Distribution` and
+ * distributes the grid points accordingly.
+ *
+ * The formula for the mapping is, in case of a `Linear` distribution
+ * \f{align}
+ * x &= \frac{b}{B-A} (\xi-A) + \frac{a}{B-A} (B-\xi)\\
+ * \xi &=\frac{B}{b-a} (x-a) + \frac{A}{b-a} (b-x)
+ * \f}
+ *
+ * which we use in each other distribution to map an interval to \f$[-1, 1]\f$
+ * with
+ * \f{align}
+ * f(\xi) &:= \frac{A+B-2\xi}{A-B} \in [-1, 1]\\
+ * g(x) &:= \frac{a+b-2x}{a-b} \in [-1, 1]
+ * \f}
+ *
+ * With this an `Equiangular` distribution is described by
+ *
+ * \f{align}
+ * x &= \frac{a}{2} \left(1-\mathrm{tan}\left(\frac{\pi}{4}f(\xi)\right)\right)
+ * + \frac{b}{2} \left(1+\mathrm{tan}\left(\frac{\pi}{4}f(\xi)\right)\right)\\
+ * \xi &=
+ * \frac{A}{2} \left(1-\frac{4}{\pi}\mathrm{arctan}\left(g(x)\right)\right) +
+ * \frac{B}{2} \left(1+\frac{4}{\pi}\mathrm{arctan}\left(g(x)\right)\right)
+ * \f}
+ *
+ * \note The equiangular distribution is intended to be used with the `Wedge`
+ * map when equiangular coordinates are chosen for those maps. For more
+ * information on this choice of coordinates, see the documentation for `Wedge`.
+ *
+ *
+ * For both the `Logarithmic` and `Inverse` distribution, we first shift the
+ * singularity out of the target interval by specifying
+ * \f$c:=\f$`singularity_pos` to get
+ *
+ * for the `Logarithmic` distribution
+ *
+ * \f{align}
+ * x &= (a-c)^{0.5 \left(1-f(\xi)\right)} (b-c)^{0.5 \left(1 + f(\xi)\right)}\\

```

Comment:
Something is off here I think. For one, xi=A gives x=a-c, so a `+c` is missing. Maybe it would be better to write this closer to the form you implemented in the code.

Reply:
I messed up and have added the +c to the equations below. Thank you for checking so diligently.